forc/src/cli/commands/new.rs

```diff
@@ -0,0 +1,35 @@
+use crate::ops::forc_new;
+use anyhow::Result;
+use clap::Parser;
+
+const TEMPLATE_HELP: &str = r#"Initialize a new project from a template.
+
+Example Templates: 
+ - counter"#;
+
+/// Create a new Forc project.
+#[derive(Debug, Parser)]
+pub struct Command {
+    /// Initialize a new project from a template
+    #[clap(short, long, help = TEMPLATE_HELP)]
+    pub template: Option<String>,
+    /// The default program type, excluding all flags or adding this flag creates a basic contract program.
+    #[clap(long)]
+    pub contract: bool,
+    /// Adding this flag creates an empty script program.
+    #[clap(long)]
+    pub script: bool,
+    /// Adding this flag creates an empty predicate program.
+    #[clap(long)]
+    pub predicate: bool,
+    /// Adding this flag creates an empty library program.
+    #[clap(long)]
+    pub library: bool,
+    /// The name of your project
+    pub project_name: String,
```

Comment:
Let's rename this field to `path` and update the comment. E.g. the following should be valid:

```
forc new my-project
```
```
forc new foo/bar/my-project
```

The name of the project should be taken from the final directory in the path (i.e. `my-project` in the above examples).

If we want to match Rust's behaviour, we could also include a `--name <NAME>` option that allows the user to specify a forc project name that doesn't necessarily match the final directory name. We can leave that for a future PR if you'd prefer though.